src/Libraries/Microsoft.Extensions.Compliance.Abstractions/Classification/DataClassificationSet.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Shared.Diagnostics;
+using Microsoft.Shared.Pools;
+
+namespace Microsoft.Extensions.Compliance.Classification;
+
+/// <summary>
+/// Represents a set of data classes.
+/// </summary>
+public sealed class DataClassificationSet : IEquatable<DataClassificationSet>
+{
+    private readonly HashSet<DataClassification> _classifications = [];
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DataClassificationSet"/> class.
+    /// </summary>
+    /// <param name="classification">The class to include in the set.</param>
+    public DataClassificationSet(DataClassification classification)
+    {
+        _ = _classifications.Add(classification);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DataClassificationSet"/> class.
+    /// </summary>
+    /// <param name="classifications">The classes to include in the set.</param>
+    public DataClassificationSet(IEnumerable<DataClassification> classifications)
+    {
+        _ = Throw.IfNull(classifications);
+        _classifications.UnionWith(classifications);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DataClassificationSet"/> class.
+    /// </summary>
+    /// <param name="classifications">The classes to include in the set.</param>
+    public DataClassificationSet(params DataClassification[] classifications)
+    {
+        _ = Throw.IfNull(classifications);
+        _classifications.UnionWith(classifications);
+    }
+
+    /// <summary>
+    /// Converts a data classification to a data classification set.
+    /// </summary>
+    /// <param name="classification">The classification to include in the set.</param>
+    public static implicit operator DataClassificationSet(DataClassification classification)
+    {
+        return FromDataClassification(classification);
+    }
+
+    /// <summary>
+    /// Converts a data classification to a data classification set.
+    /// </summary>
+    /// <param name="classification">The classification to include in the set.</param>
+    /// <returns>The resulting data classification set.</returns>
+    public static DataClassificationSet FromDataClassification(DataClassification classification) => new(classification);
+
+    /// <summary>
+    /// Creates a new data classification that combines the current classifications with another set.
+    /// </summary>
+    /// <param name="other">The other set.</param>
+    /// <returns>The resulting set which combines the current instance's classifications and the other ones.</returns>
```

Comment:
Maybe put a remark that this function doesn't change the current set and returns a new copy with the result?

Reply:
Done.